src/Essentials/src/MainThread/MainThread.uwp.cs

```diff
@@ -6,17 +7,21 @@ namespace Microsoft.Maui.ApplicationModel
 	public static partial class MainThread
 	{
 		static bool PlatformIsMainThread =>
-			WindowStateManager.Default.GetActiveWindow().DispatcherQueue.HasThreadAccess;
+			TryGetDispatcherQueue()?.HasThreadAccess ?? false;
 
 		static void PlatformBeginInvokeOnMainThread(Action action)
 		{
-			var dispatcher = WindowStateManager.Default.GetActiveWindow().DispatcherQueue;
+			var dispatcher = TryGetDispatcherQueue();
 
 			if (dispatcher == null)
 				throw new InvalidOperationException("Unable to find main thread.");
 
 			if (!dispatcher.TryEnqueue(DispatcherQueuePriority.Normal, () => action()))
 				throw new InvalidOperationException("Unable to queue on the main thread.");
 		}
+
+		static DispatcherQueue? TryGetDispatcherQueue() =>
+			DispatcherQueue.GetForCurrentThread() ??
```

Comment:
GetForCurrentThread - this is get for current _UI/Main_ thread? is that assumed?

Reply:
This gets for the current thread. So if you are in the background, it will return null. However, at startup, you are on the thread with a dispatcher. But yeah, no way to get the "main thread" as any thread can be a "main thread" if it has a UI context set up.